test/parallel/test-readline-line-separators.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+const common = require('../common');
+const assert = require('node:assert');
+const readline = require('node:readline');
+const { Readable } = require('node:stream');
+
+const str = '123\n456\r789\u{2028}ABC\u{2029}DEF';
+
+const rli = new readline.Interface({
+  input: Readable.from(str),
+});
+
+const linesRead = [];
+rli.on('line', (line) => linesRead.push(line));
+
+rli.on('close', common.mustCall(() => {
+  const regexpLines = str.split(/^/m).map((line) => line.trim());
+  // Readline interprets different lines in the same way js regular expressions do
+  assert.deepStrictEqual(linesRead, regexpLines);
```

Comment:
The assertion fails if `str` contains `\r\n`.

Reply:
Thanks! :slightly_smiling_face: (also it is my bad for not including `\r\n` in the test :sweat:)

Fixed :+1: 

Although I added a `filter(Boolean)` to the `regexpLines`, so the comment `Readline interprets different lines in the same way js regular expressions do` might be slightly is not 100% correct anymore :thinking: , what do you think?

(PS: maybe I should just not compare it to the result of the regexp splitting but add the expected array in the test instead :thinking:)